src/tools/storage-key-recipe-resolver.ts

```diff
@@ -95,17 +95,15 @@ export class StorageKeyRecipeResolver {
    * @param recipe should be long running.
    * @param arc Arc is associated with current recipe.
    */
-  async createStoresForCreateHandles(recipe: Recipe, arc: Arc) {
-    const dbKeyCreator: StorageKeyCreatorInfo = {
-      protocol: 'db',
-      capabilities: Capabilities.persistentQueryable,
-      create: options => new PersistentDatabaseStorageKey(options.unique(), options.schemaHash),
-    };
-    const resolver = new CapabilitiesResolver({arcId: arc.id},
-      [dbKeyCreator, ...CapabilitiesResolver.getDefaultCreators()]);
-    for (const createHandle of recipe.handles.filter(h => h.fate === 'create' && !!h.id)) {
+  async createStoresForCreateHandles(recipe: Recipe, arc: Arc): Promise<Recipe> {
+    DatabaseStorageKey.register();
+    const resolver = new CapabilitiesResolver({arcId: arc.id});
+    if (recipe.handles.filter(h => h.fate === 'create' && !!h.id).length === 0) {
```

Comment:
Let's save the result of the filter into a variable, so that the maintainer won't make the mistake of changing one without the other.

Reply:
Hmm... I see that the second filter is of a clone. I see why this logic is repeated.

Reply:
i decided to remove the `if` statement and just always clone the recipe to avoid this confusion.